frontend/lib/qr_code_scanner/qr_code_scanner_controls.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:async';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+
+class QRCodeScannerControls extends StatelessWidget {
+  final QRViewController controller;
+
+  // Not really nice. We need to update the button texts after we pressed them.
+  // Other possibility would be to convert this to a stateful widget without
+  // state and just call setState({}) to trigger redrawing of whole widget.
+  final StreamController<bool> flashStreamController = StreamController();
+  final StreamController<CameraFacing> cameraStreamController =
+    StreamController<CameraFacing>();
+
+  QRCodeScannerControls({
+    @required this.controller,
+    Key key
+  }) : super(key: key) {
+    updateFlashStream();
+    updateCameraStream();
+  }
+
+  Future<void> updateFlashStream() =>
+      controller.getFlashStatus().then(flashStreamController.add);
+  Future<void> updateCameraStream() =>
+      controller.getCameraInfo().then(cameraStreamController.add);
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<SystemFeatures>(
+        future: controller.getSystemFeatures(),
+        builder: (context, snapshot) =>
+      Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: <Widget>[
+            if (snapshot.data.hasFlash)
+              Container(
```

Comment:
On iOS in the qr_code_scanner_controls.dart, the first call on `controller.getSystemFeatures()` causes a `CameraException(404, null)`. From testing on my ipad, it takes about 60ms until the first request succeeds.

One solution could be to wrap `controller.getSystemFeatures()` with a retry mechanism like this:
```dart
  Future<SystemFeatures> tryToGetSystemFeatures() async {
    for (var i = 0; i < 10; i++) {
      await Future.delayed(Duration(milliseconds: 100));
      try {
        return await controller.getSystemFeatures();
      } on CameraException {
        continue;
      }
    }
    return null;
  }
```

A hardcoded delay on ios with like 100ms might work, too, but i dont know if it might take longer than 100ms on some older ios devices.

Reply:
Oh man 😄 I added your solution with the retry. Could you test it again?

If it still does not work, we can just throw this system feature check away and just always display both buttons as before. Found it a quick win, but did not expect more of these strange iOS bugs 😄

Reply:
Hm, found at least two other bugs when scanning random qr codes... I'll try to fix them

Reply:
Wow, seems like the base32 encoding library we use crashes the thread when given a null value which happened in my test. The interesting thing is that there is no exception that can be caught, also the debugger sees nothing. The thread just exits silently (but the app keeps running). Wtf flutter/dart